packages/edit-site/src/components/global-styles/font-library-modal/utils/index.js

```diff
@@ -135,39 +136,84 @@ export function getDisplaySrcFromFontFace( input, urlPrefix ) {
 	return src;
 }
 
-export function makeFormDataFromFontFamily( fontFamily ) {
+export function makeFontFamilyFormData( fontFamily ) {
 	const formData = new FormData();
 	const { kebabCase } = unlock( componentsPrivateApis );
 
-	const newFontFamily = {
-		...fontFamily,
+	const { fontFace, category, ...familyWithValidParameters } = fontFamily;
+	const fontFamilySettings = {
+		...familyWithValidParameters,
 		slug: kebabCase( fontFamily.slug ),
 	};
 
-	if ( newFontFamily?.fontFace ) {
-		const newFontFaces = newFontFamily.fontFace.map(
-			( face, faceIndex ) => {
-				if ( face.file ) {
-					// Slugified file name because the it might contain spaces or characters treated differently on the server.
-					const fileId = `file-${ faceIndex }`;
-					// Add the files to the formData
-					formData.append( fileId, face.file, face.file.name );
-					// remove the file object from the face object the file is referenced by the uploadedFile key
-					const { file, ...faceWithoutFileProperty } = face;
-					const newFace = {
-						...faceWithoutFileProperty,
-						uploadedFile: fileId,
-					};
-					return newFace;
-				}
-				return face;
+	formData.append(
+		'font_family_settings',
+		JSON.stringify( fontFamilySettings )
+	);
+	return formData;
+}
+
+export function makeFontFacesFormData( font ) {
+	if ( font?.fontFace ) {
+		const fontFacesFormData = font.fontFace.map( ( face, faceIndex ) => {
+			const formData = new FormData();
+			if ( face.file ) {
+				// Slugified file name because the it might contain spaces or characters treated differently on the server.
+				const fileId = `file-${ faceIndex }`;
+				// Add the files to the formData
+				formData.append( fileId, face.file, face.file.name );
+				// remove the file object from the face object the file is referenced in src
+				const { file, ...faceWithoutFileProperty } = face;
+				const fontFaceSettings = {
+					...faceWithoutFileProperty,
+					src: fileId,
+				};
+				formData.append(
+					'font_face_settings',
+					JSON.stringify( fontFaceSettings )
+				);
+			} else {
+				formData.append( 'font_face_settings', JSON.stringify( face ) );
 			}
-		);
-		newFontFamily.fontFace = newFontFaces;
+			return formData;
+		} );
+
+		return fontFacesFormData;
 	}
+}
 
-	formData.append( 'font_family_settings', JSON.stringify( newFontFamily ) );
-	return formData;
+export async function batchInstallFontFaces( fontFamilyId, fontFacesData ) {
+	const promises = fontFacesData.map( ( faceData ) =>
+		fetchInstallFontFace( fontFamilyId, faceData )
+	);
+	const responses = await Promise.allSettled( promises );
```

Comment:
Why was `allSettled` used over `all` here?

This should be wrapped in a try / catch

Reply:
> Why was `allSettled` used over `all` here?

We’re using `allSettled` because `all` will reject as soon as one fails. `allSettled` will wait for each one to be completed, fulfilled, or rejected. In our use case, it’s possible to get into a state where there are some successful and some failed font face installations, and we want to know the details so we can handle the activation and error reporting accordingly.

> This should be wrapped in a try / catch

This function is already being wrapped in a try/catch block where it's being called in `installFont` [here](https://github.com/WordPress/gutenberg/pull/57688/files#diff-ee8604b8acd83e45973351b79dc6983dc4daad84903a30a41e105a3d7429b74bR247), where we can handle the error messages on a more detailed level.